backend/auth.py

```diff
@@ -243,3 +249,58 @@ def handle_request(response):
 
         redirect = self.get_argument("next", '/')
         self.redirect(redirect)
+
+
+class UpdateUserHandler(BaseHandler):
+    def get(self):
+        """
+        If a user is logged in to elixir, and also has google login cookies, the
+        google users information in the database will be updated with the elixir
+        users information.
+        """
+        try:
+            # Double check so that the elixir user isn't already have any credentials
+            # in the database.
+
+            elixir_email = self.get_secure_cookie('email')
+
+            (db.User.select()
+                    .join(db.DatasetAccess)
+                    .where(
+                           db.User.user == db.DatasetAccess.user,
+                           db.User.email == elixir_email)
```

Comment:
Compare the `db.User.identity` field to the `identity` cookie instead of email. The email we get from elixir can change for the same user, but their `identity` (or `sub`) should stay the same.